### PR TITLE
Feature/37

### PR DESCRIPTION
**Pull request by enicolasgomez**
as requested in #37 

created `removeDisconnectedObjects` taking two parameters as arguments: the previous model and the connectivity model, the function will do the following:

- extract all the `wirtedTo` ids from the connectivityModel's chains
- extract all the 'hostedBy' ids from the original model anchors
- extract all the 'hostedBy' ids from the connectivityModel's chains
- filter any regions or wires which are not included in any of the previous lists

This is how the result looks like:

image

### CODE REVIEW

File: src/components/webGLScene.js [hunk 1]
```diff
@@ -531,7 +531,8 @@ export class WebGLSceneComponent {
 
     getMouseOverEntity() {
         if (!this.graph) { return; }
-        this.ray.setFromCamera( this.mouse, this.camera );
+        const _3dmouse = { ...this.mouse, z : 0.5 }
```
Comment (zsinnema) [inline src/components/webGLScene.js:534]: @enicolasgomez please explain why this is needed

File: src/test-app/app.js [hunk 2]
```diff
@@ -371,6 +371,23 @@ export class TestApp {
         }
     }
 
+    removeDisconnectedObjects(model, joinModel) {
+      let connected = joinModel.chains.map((c)=> { return c.wiredTo }).filter((c) => { return c !== undefined });
```
Comment (zsinnema) [inline src/test-app/app.js:375]: can be written as
```
let connected = joinModel.chains.map((c)=> c.wiredTo).filter((c) => c !== undefined);
```

File: src/test-app/app.js [hunk 3]
```diff
@@ -371,6 +371,23 @@ export class TestApp {
         }
     }
 
+    removeDisconnectedObjects(model, joinModel) {
+      let connected = joinModel.chains.map((c)=> { return c.wiredTo }).filter((c) => { return c !== undefined });      
+      connected = [ ...connected, ... model.anchors.map((c)=> { return c.hostedBy }).filter((c) => { return c !== undefined }) ];
```
Comment (zsinnema) [inline src/test-app/app.js:376]: same here, if you are only returningvalues  then please don't specify the `{ return }`

File: src/test-app/app.js [hunk 4]
```diff
@@ -371,6 +371,23 @@ export class TestApp {
...
+
+      const hostedBy = joinModel.chains.filter((l) => { return l.hostedBy != undefined } ).map((f) => { return f.hostedBy });
```
Comment (zsinnema) [inline src/test-app/app.js:378]: just a hint....
when lines become long then they become unreadable, split them over lines

File: src/test-app/app.js (hunk 3 above)
Comment (zsinnema) [reply src/test-app/app.js:376]: why create a new array (use of spread) if you can also use `concat` and in that case connected can be a const

File: src/test-app/app.js [hunk 5]
```diff
@@ -394,12 +411,14 @@ export class TestApp {
             throw new Error("Cannot join models with the same identifiers: " + this._model.id);
         }
         if (isScaffold(this._model) !== isScaffold(newModel)){
-            this.applyScaffold(this._model, newModel);
+          this.model = this.removeDisconnectedObjects(this._model, newModel);
```
Comment (zsinnema) [inline src/test-app/app.js:414]: question: `this.model` or `this._model` what is the difference between them?

**Review by zsinnema — CHANGES_REQUESTED** [review]

File: src/components/webGLScene.js (hunk 1 above)
Comment (enicolasgomez) [reply src/components/webGLScene.js:534]: Not needed. That's a left over while investigating an other issue. Will remove it.

**Review by zsinnema — APPROVED** [review]

**Pull request merged**